docs/how-to/broker-priority.md

```diff
@@ -0,0 +1,83 @@
+Their resource consumption depends on the number of cluster nodes and number of pods running in total. 
+
+As existing components use more resources and as we add more new components, more cluster and node resources
+are required to run Kubernetes. This [summary](https://github.com/google/knative-gcp/issues/1502#issuecomment-664793074)  
+illustrates the broker components may at risk to be OOMKilled/evicted/preempted when node is over-committed. 
+
+To protest broker components from eviction or preemption, we would like to give them higher scheduling/eviction priority.
+However, this would put other applications at risk in the case of insufficient resource. To balance the needs, 
+we have a scaling mechanism for broker components, and an on-going work to provide configurable resource request and limit.
+
+Beside scaling and config, users are always able to set up pod priority for the broker components 
+if they regard the broker components a high priority components and would like to further reduce its possibility
+to be the victim of preemption. Follow this doc to set up a Pod Priority for your broker components.
+
+## Add PriorityClass for broker
+
+Apply the following yaml to add a [PriorityClass](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#priorityclass) for the broker:
+
+```shell
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: broker-priority
+value: 1000
+preemptionPolicy: Never
+globalDefault: false
+description: "Priority class for broker. This priority class will not cause other pods to be preempted."
+```
+
+Here, we set `PreemptionPolicy: Never` to avoid broker components [preempting other pods](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#non-preempting-priority-class).
+
+There is no certain criteria to choose the priority value. You can choose a value which you think is correctly reflecting
+the importance of the broker components. 
+
+Pods without PriorityClass specified have the default Priority value: `0`.
+Pods running critically for the cluster or for the node are generally with a priority value bigger than `1000000000`(one billion).
+
+
+## Update broker related deployment with PriorityClass
+
+1) Make sure you have at least one broker in your cluster.
+2) Apply the following command to check the
+    broker data plane:
+    ```
+    kubectl get deployment -n cloud-run-events
+    ```
+    
+    You are going to update `deployment`s:
+    
+    ```
+    NAME                         READY   UP-TO-DATE   AVAILABLE   AGE
+    default-brokercell-fanout    1/1     1            1           4h59m
+    default-brokercell-ingress   1/1     1            1           4h59m
+    default-brokercell-retry     1/1     1            1           4h59m
+    ```
+3) Add `PriorityClass` to each deployment. This field is located at `spec.template.spec` of the `deployment`, looks like:
```

Comment:
It might be useful to note that this change will cause a new pod to be created, and the existing pod to be deleted after the new pod is available.

Reply:
Done.